File: 2023/day_1-trebuchet/day_1-trebuchet.ts
import fs from "fs";

const file = fs.readFileSync(__dirname + "/input.txt").toString();

const lines = file.trim().split("\n");

let sum = 0;

for (const line of lines) {
    let first_digit: string = "0";
    for (let i = 0; i < line.length; i++) {
        const c = line[i];
        if ("0" <= c && c <= "9") {
            first_digit = c;
            break;
        }
    }
    let last_digit: string = "0";
    for (let i = line.length - 1; i >= 0; i--) {
        const c = line[i];
        if ("0" <= c && c <= "9") {
            last_digit = c;
            break;
        }
    }
    const num = Number(first_digit + last_digit);

    sum += num;
}

console.log(sum);

File: 2023/day_1-trebuchet/day_1-trebuchet.go
package main

import (
	"log"
	"os"
	"strings"
)

func main() {
	c, err := os.ReadFile("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	file := strings.Trim(string(c), "")
	lines := strings.Split(file, "\n")

	sum := 0

	for _, line := range lines {
		fd := '0'
		for _, c := range line {
			if '0' <= c && c <= '9' {
				fd = c
			}
		}
		ld := '0'
		for _, c := range line {
			if '0' <= c && c <= '9' {
				fd = c
			}
		}
	}
}
